confidence/models.py

```diff
@@ -30,7 +30,7 @@ class Configuration(Mapping):
     """
 
     # match a reference as ${key.to.be.resolved}
-    _reference_pattern = re.compile(r'\${(?P<path>[^}]+?)}')
+    _reference_pattern = re.compile(r'\${(?P<path>[^${}]+?)}')
```

Comment:
Actual fix here: match the shortest path from `${` to `}` *not* containing any of those particular characters. Previously, `${reference.${within}.value}` would match `${reference.${within}` as a single reference first, which wouldn't work. New pattern will match `${within}` first, resolve it and iteratively carry on as before.